src/runner/observer.rs

```diff
@@ -1,19 +1,18 @@
 //! Events are sent by the Runner to signal the progression in the test suite, with the results
 
-use std::io;
-
 use header::SuiteHeader;
 use header::ContextHeader;
 use header::ExampleHeader;
 use report::SuiteReport;
 use report::ContextReport;
 use report::ExampleReport;
 
-pub trait EventHandler: Send + Sync {
-    fn enter_suite(&self, suite: &SuiteHeader) -> io::Result<()>;
-    fn exit_suite(&self, suite: &SuiteHeader, report: &SuiteReport) -> io::Result<()>;
-    fn enter_context(&self, context: &ContextHeader) -> io::Result<()>;
-    fn exit_context(&self, context: &ContextHeader, _report: &ContextReport) -> io::Result<()>;
-    fn enter_example(&self, example: &ExampleHeader) -> io::Result<()>;
-    fn exit_example(&self, example: &ExampleHeader, report: &ExampleReport) -> io::Result<()>;
+/// `RunnerObserver`s can be attached to a [`Runner`](../runner/struct.Runner.html) to observe a
+pub trait RunnerObserver: Send + Sync {
```

Comment:
A default empty impl for each would be nice, here

Reply:
it would remove a lot of boilerplate code

Reply:
Yeah, good idea!